chain/deals/client_utils.go

```diff
@@ -98,3 +109,81 @@ func (c *Client) disconnect(deal ClientDeal) error {
 	delete(c.conns, deal.ProposalCid)
 	return err
 }
+
+var _ datatransfer.RequestValidator = &ClientRequestValidator{}
+
+// ClientRequestValidator validates data transfer requests for the client
+// in a storage market
+type ClientRequestValidator struct {
+	deals *statestore.StateStore
+}
+
+// RegisterClientValidator is an initialization hook that registers the client
+// request validator with the data transfer module as the validator for
+// StorageDataTransferVoucher types
+func RegisterClientValidator(lc fx.Lifecycle, crv *ClientRequestValidator, dtm dtypes.ClientDataTransfer) {
```

Comment:
I'd prefer if we can keep the fx code out of the implementation. We could just expose an 'OnStart' method on the ClientDataTransfer module